sdv/single_table/copulas.py

```diff
@@ -83,18 +83,29 @@ def get_distribution_class(cls, distribution):
 
         return cls._DISTRIBUTIONS[distribution]
 
+    def _validate_numerical_distributions(self, numerical_distributions):
+        if numerical_distributions:
+            if not isinstance(numerical_distributions, dict):
+                raise TypeError('numerical_distributions can only be None or a dict instance.')
+
+            invalid_columns = numerical_distributions.keys() - set(self.metadata._columns)
+            if invalid_columns:
+                raise SynthesizerInputError(
+                    'Invalid column names found in the numerical_distributions dictionary '
+                    f'{invalid_columns}. The column names you provide must be present '
+                    'in the metadata.'
+                )
+
```

Comment:
Lets convert this into a `staticmethod` that takes `columns` and `numerical_distributions`, then we can re-use it in the `copulagan` by using `GaussianCopulaSynthesizer._validate_numerical_distributions(...)` and not having the code duplicated.